### PR TITLE
[Doc] Improve documentation of disabling annotations in pure mode

### PR DESCRIPTION
**Pull request by matusvalo**

### CODE REVIEW

**Comment by matusvalo** [comment]
> I missed that there is also `@cython.annotation_typing()` decorator. It is not documented, so I will add it to this PR.

Done. Ready for review

File: docs/src/tutorial/pure.rst [hunk 1]
```diff
@@ -378,6 +377,19 @@ declare types of variables in a Python 3.6 compatible way as follows:
 
 There is currently no way to express the visibility of object attributes.
 
+Disabling annotations
+^^^^^^^^^^^^^^^^^^^^^
+
+To avoid conflicts with other kinds of annotation
+usages, reading of annotation by Cython can be disabled with the ``@cython.annotation_typing(False)`` decorator:
```
Comment (da-woods) [inline docs/src/tutorial/pure.rst:384]:
```suggestion
usages, Cython's use of annotations to specify types can be disabled with the
``annotation_typing`` compiler directive<compiler-directives>. From Cython 3
you can use this as a decorator, as shown in the following example:
```

File: docs/src/tutorial/pure.rst [hunk 2]
```diff
@@ -378,6 +377,19 @@ declare types of variables in a Python 3.6 compatible way as follows:
...
+
+.. literalinclude:: ../../examples/tutorial/pure/disabled_annotations.py
+
+Another way to disable annotations is by using :ref:`compiler directive<compiler-directives>` ``annotation_typing=False``::
+
+    cython -X annotation_typing=False ignored_annotations.py
```
Comment (da-woods) [inline docs/src/tutorial/pure.rst:390]:
```suggestion
```

**Review by da-woods — COMMENTED** [review]
I've rewritten it slightly, but don't feel you have to use my exact wording. The changes I made were:

* I think it's the same compiler directive whether you use it as a decorator, or a `with` statement, or at a file level,
* I think it's worth making clear that the ability to use it as a decorator is a Cython 3 improvement.

**Comment by matusvalo** [comment]
> or a with statement

`cython.annotation_typing()` supports also `with` statement? If yes I prefer to mention it in docs too...

**Comment by da-woods** [comment]
> cython.annotation_typing() supports also with statement? If yes I prefer to mention it in docs too...

Yes it should do. I don't think it needs special mention in the text - most compiler directives can be used as a `with` statement. You could definitely include it in your example though - it'd let you skip typing one or two variables.

File: docs/examples/tutorial/pure/disabled_annotations.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+
+def annotated_function(a: cython.int, b: cython.int):
+    s: cython.int = a + b
+    with cython.annotation_typing(False):
+        # Cython is ignoring annotations within this code block
+        c: list = []
+        c.append(a)
+        c.append(b)
+        c.append(s)
```
Comment (da-woods) [inline docs/examples/tutorial/pure/disabled_annotations.py:32]: If I were writing this I'd only put the `c: list` in the `with` block. But either way will compile

Comment (matusvalo) [reply docs/examples/tutorial/pure/disabled_annotations.py]: > If I were writing this I'd only put the `c: list` in the `with` block. But either way will compile

fixed.

**Comment by matusvalo** [comment]
Please have a final look.

**Comment by da-woods** [comment]
Just waiting to make sure that the CI is happy with the code.... looks good otherwise

**Comment by matusvalo** [comment]
> Just waiting to make sure that the CI is happy with the code.... looks good otherwise

btw have you tried auto-merge?
image

**Comment by matusvalo** [comment]
Hmm CI fail seems to be unrelated to this PR...

**Comment by da-woods** [comment]
Thanks - I think it's useful to have this information up front since it Cython won't always do what people want with annotations

**Pull request merged**